File: AVc/app/src/main/java/com/hfad/avc/ui/list/all_congratulations_list/AllCongratulationsListPresenter.java
package com.lastaurus.automatic_congratulations.ui.list.all_congratulations_list;

import com.lastaurus.automatic_congratulations.dagger.ComponentManager;
import com.lastaurus.automatic_congratulations.managers.DBManager;

import javax.inject.Inject;

import moxy.InjectViewState;
import moxy.MvpPresenter;

@InjectViewState
public class AllCongratulationsListPresenter extends MvpPresenter<IAllCongratulationsListViewModel> {

   @Inject
   DBManager dbManager;

   public AllCongratulationsListPresenter() {
      ComponentManager.Companion.getInstance().appComponent.inject(this);
   }

   @Override
   public void attachView(IAllCongratulationsListViewModel view) {
      super.attachView(view);
      getViewState().setData(dbManager.getContactList(1));
   }
}

File: app/src/main/java/com/lastaurus/automatic_congratulations/ui/list/all_congratulations_list/AllCongratulationsListPresenter.kt
package com.lastaurus.automatic_congratulations.ui.list.all_congratulations_list

import com.lastaurus.automatic_congratulations.dagger.ComponentManager.Companion.instance
import moxy.InjectViewState
import moxy.MvpPresenter


@InjectViewState
class AllCongratulationsListPresenter :
   MvpPresenter<IAllCongratulationsListViewModel>() {
   init {
      instance.appComponent.inject(this)
   }

   override fun attachView(view: IAllCongratulationsListViewModel) {
      super.attachView(view)
      viewState?.setData()
   }
}